### ISSUE TITLE
Chaos: Terminating Gateway Stops Workflow Processing 

### ISSUE THREAD

**Issue by shahamit** [zeebe-io/zeebe-chaos#336]
## Chaos Experiment
When running the terminate chaos experiment against a zeebe cluster that was under load, we observed that the cluster stops processing any workflows there after.

Config - 6 brokers, 2 gateways, 6 partitions, 2 replication factor.

Note we don't have an ingress controller configured in front of the zeebe-gateway. Since our client (benchmarking tool in this case) runs within the same zeebe cluster it should be fine given that k8s service (zeebe-gateway) does the load balancing between them (which isn't happening but that's a separate issue). 

We were hoping that since the client (benchmarking tool) connects to the k8s zeebe-gateway service, terminating one of the gateway instances shouldn't have any impact on the client. I didn't follow why do we see errors on the client. Please share more insights.

Thanks.

**Benchmarking tool logs**
![ksnip_20230321-181948](https://user-images.githubusercontent.com/1281926/226633647-4303486f-db69-48eb-9af3-a671c82d32f0.png)

**Terminate command output**
![ksnip_20230321-181029](https://user-images.githubusercontent.com/1281926/226633558-292e5eb8-29b7-4284-b05c-d200e6dac936.png)

**Comment by Zelldon** [comment]
>We were hoping that since the client (benchmarking tool) connects to the k8s zeebe-gateway service, terminating one of the gateway instances shouldn't have any impact on the client. 

The affect will never be zero, because some request might fail or timeout, but yes after retry it should work and take the next gateway I agree.

**Comment by shahamit** [comment]
> What means it stops, does it recover afterwards? After some time eventually?

Yes after a few seconds, the workflows did start getting processed. In between though some workflow do fail (indicated by the backpressure % increasing). Given that the gateway replicas are behind a k8s service, shouldn't it automatically go to the next gateway instance instead of failing the workflows?

**Comment by Zelldon** [comment]
Do you have any metrics to show? What type of load we are speaking of? :thinking: 

> Given that the gateway replicas are behind a k8s service, shouldn't it automatically go to the next gateway instance 

Yes if a new request comes in I would expect something like that.

**Comment by shahamit** [comment]
Sorry for the late reply @Zelldon 

> Do you have any metrics to show? What type of load we are speaking of? thinking

We are running the [benchmarking tool](https://github.com/camunda-community-hub/camunda-8-benchmark) against a zeebe cluster of 7 brokers and 2 gateways. We could see the throughput as 170 PI/s. 

> Be aware that the process instances are not failing, they are just not continued right?

This is hard to find out since the benchmarking tool starts around 170 process instances per second. If you can think of a way to find this out, please let me know.

Thanks
